services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -578,7 +583,25 @@ def _compute_statistics(
         df_without_na = data.select(pl.col(column_name)).drop_nulls()
 
         lengths_column_name = f"{column_name}_len"
-        lengths_df = df_without_na.with_columns(pl.col(column_name).list.len().alias(lengths_column_name))
+        column_schema = data.schema[column_name]
+        if isinstance(column_schema, List):
+            lengths_df = df_without_na.with_columns(pl.col(column_name).list.len().alias(lengths_column_name))
+
+        elif isinstance(column_schema, Struct):
+            if not all(isinstance(field.dtype, List) for field in column_schema.fields):
+                raise PolarsSequenceSchemaError(
+                    f"Error parsing {column_name=} {column_schema=}. Subfields of Struct must be Lists. "
+                )
+            first_field_name = column_schema.fields[0].name  # assuming all fields have equal lengths within a sample
+            lengths_df = df_without_na.with_columns(
+                pl.col(column_name).struct[first_field_name].list.len().alias(lengths_column_name)
+            )
+        else:
```

Comment:
idk if there might be something else (except for List and Struct), i assume that no but we can check after deployment if there are errors like this